tart2ms/tart2ms.py

```diff
@@ -355,21 +377,52 @@ def __twelveball(direction):
-        direction = np.array([0, np.deg2rad(-90)]).reshape(1, 2, 1)
-    elif phase_center_policy == "rephase-NCP":
-        direction = np.array([0, np.deg2rad(90)]).reshape(1, 2, 1)
+    elif phase_center_policy.find("rephase-") == 0:
+        fn = phase_center_policy.replace("rephase-","")
+        use_special_fn = fn
+        # 3CRR currently only catalog with special names
+        catalog_positions = catalog_reader.catalog_factory.from_3CRR(fluxlim15=0.0)
+        named_positions = read_known_phasings()
+        fsrc = list(filter(lambda x: x.name == fn, catalog_positions))
+        if len(fsrc) > 0:
+            direction = np.array([[fsrc[0].rarad, fsrc[0].decrad]]).reshape(1, 2, 1)
+        else: # otherwise it is in the named positions list
+            fsrc = list(filter(lambda x: x['name'].upper() == fn, named_positions))
+            if len(fsrc) == 0:
+                raise RuntimeError(f"Unknown named source {fn}")
+            if fsrc[0]['position']['FRAME'] == "Special Body":
+                body = ac.get_body(fn, obstime[0], location=location)
+                LOGGER.critical(f"User caution: enabling non-sidereal tracking of '{fn}'. Indicated field centre will be "
+                                f"that of the first timestamp in your synthesized map, but W-coordinate will be actively "
+                                f"fringestopping at source position")
+                direction = np.deg2rad(np.array([[body.icrs.ra.value, body.icrs.dec.value]]).reshape(1, 2, 1))
+            else:
+                # we assume we can parse it with astropy
+                ra = fsrc[0]['position']["RA"]
+                dec = fsrc[0]['position']["DEC"]
+                equinox = fsrc[0]['position']["EQUINOX"]
+                frame = fsrc[0]['position']["FRAME"]
+                crd = SkyCoord(f"{ra} {dec}", equinox=equinox, frame=frame)
+                direction = np.deg2rad(np.array([[crd.icrs.ra.value, crd.icrs.dec.value]]).reshape(1, 2, 1))
     else:
         raise ValueError(f"phase_center_policy must be one of "
-                         f"['instantaneous-zenith','rephase-obs-midpoint','rephase-SCP','rephase-NCP',"
-                         f"'no-rephase-obs-midpoint'] got {phase_center_policy}")
-    field_name = da.asarray(np.array(list(map(__twelveball, direction.reshape(2, direction.shape[2]).T)),
-                                     dtype=object), chunks=1)
+                         f"['instantaneous-zenith','rephase-obs-midpoint','no-rephase-obs-midpoint',"
+                         f"'rephase-<named position>' or Astropy.SkyCoord] got {phase_center_policy}")
+    if use_special_fn:
+        field_name = da.asarray(np.array([use_special_fn] * direction.shape[2]),
+                                dtype=object, chunks=1)
+    else:
+        field_name = da.asarray(np.array(list(map(__twelveball, direction.reshape(2, direction.shape[2]).T)),
+                                         dtype=object), chunks=1)
```

Comment:
Named positions (whether from catalog or the json e.g. solar system bodies) will be named. The rest e.g. obs-midpoint phasing will result in field names with JXXXXXX-+XXXXXX convention